utilities/rocAL/rocal_external_source/external_source.py

```diff
@@ -0,0 +1,123 @@
+
+import sys
+import types
+import collections
+import cv2
+import numpy as np
+from random import shuffle
+from enum import Enum
+import torch
+from amd.rali.plugin.pytorch import RALIClassificationIterator
+from amd.rali.plugin.pytorch import RALI_iterator
+from amd.rali.pipeline import Pipeline
+import amd.rali.ops as ops
+import amd.rali.types as types
+
+
+class ExternalInputIterator(object):
+    def __init__(self, path, sinset_file_name, batch_size, mode, device_id, num_gpus):
+        self.images_dir = path
+        self.batch_size = batch_size
+        self.mode = mode
+        with open(self.images_dir + sinset_file_name, 'r') as f:
+            self.files = [line.rstrip() for line in f if line is not '']
+        # whole data set size
+        self.data_set_len = len(self.files) 
+        # based on the device_id and total number of GPUs - world size
+        # get proper shard
+        self.files = self.files[self.data_set_len * device_id // num_gpus:
+                                self.data_set_len * (device_id + 1) // num_gpus]
+        self.n = len(self.files)
+
+    def __iter__(self):
+        self.i = 0
+        shuffle(self.files)
+        return self
+
+    def __next__(self):
+        batch = []
+        labels = []
+
+        if self.i >= self.n:
+            self.__iter__()
+            raise StopIteration
+
+        for _ in range(self.batch_size):
+            jpeg_filename, label = self.files[self.i % self.n].split(' ')
+            if self.mode == 'file_name':
+              batch.append(self.images_dir + jpeg_filename)  # we can use numpy
+            else:
+              batch.append(np.fromfile(self.images_dir + jpeg_filename, dtype = np.uint8))  # we can use numpy
+            labels.append(torch.tensor([int(label)], dtype = torch.uint8)) # or PyTorch's native tensors
+            self.i += 1
+        return (batch, labels)
+
+    def __len__(self):
+        return self.data_set_len
+
+    next = __next__
+
+class ExternalSourcePipe(Pipeline):
+	def __init__(self, batch_size, num_threads, device_id, data_dir, file_name, image_size, rali_cpu = True):
+		super(ExternalSourcePipe, self).__init__(batch_size, num_threads, device_id, seed=12 + device_id,rali_cpu=rali_cpu)
+		world_size = 1
+		local_rank = 0
+		self.input = ops.FileReader(file_root=data_dir, shard_id=local_rank, num_shards=world_size, random_shuffle=True)
+		rali_device = 'cpu' if rali_cpu else 'gpu'
+		decoder_device = 'cpu' if rali_cpu else 'mixed'
+		device_memory_padding = 211025920 if decoder_device == 'mixed' else 0
+		host_memory_padding = 140544512 if decoder_device == 'mixed' else 0
+		self.decode = ops.ImageDecoderRandomCrop(device=decoder_device, output_type=types.RGB,
+													device_memory_padding=device_memory_padding,
```

Comment:
Please check the alignment of params in this file

Reply:
Removed this file as its not necessary